### DISCUSSION TITLE
About the C++ function ---- WaitForFrame();

### DISCUSSION THREAD

**Discussion by francisdu** [ifm/ifm3d#418] [Q&A]
I have found that the function WaitForFrame() in [frame_grabber.h](https://github.com/ifm/ifm3d/blob/legacy/modules/framegrabber/include/ifm3d/fg/frame_grabber.h) is different from it in [ifm3d::FrameGrabber Class Reference](https://api.ifm3d.com/stable/cpp_api/classifm3d_1_1FrameGrabber.html).

one is 
![image](https://github.com/ifm/ifm3d/assets/20570270/b26c5315-a98e-4efe-95e1-61c63922d4c3)
, I found it in docker image "ghcr.io/ifm/ifm3d:stable". 

And another is 
![image](https://github.com/ifm/ifm3d/assets/20570270/e1f6af54-bdae-4f43-9cb4-d75eb118b061)
.

Which one should I use?

**Comment by francisdu** [comment]
Correct: I found "std::shared_future<Frame::Ptr> ifm3d::FrameGrabber::WaitForFrame()" in docker image "ghcr.io/ifm/ifm3d:stable".

**Comment by lola-masson** [comment]
Hi @francisdu,

Which device are you using? O3R, O3D, O3X? We have some examples on how to use the library in the documentation that might be helpful to you: https://api.ifm3d.com/stable/examples/o3r/getting_data/getting_data.html. These examples are tailored to the O3R platform so you might have a couple changes to do if you are using another type of device.
